lib/resources/collection/index.js

```diff
@@ -571,7 +571,7 @@ Collection.prototype.save = function (ctx, fn) {
     if(errs) return done({errors: errs});
 
     // generate id before event listener
-    item.id = store.createUniqueIdentifier();
+    if (!item.id) item.id = store.createUniqueIdentifier();
```

Comment:
without this falsy check, item.id will always be a uuid.

It enables the post event to set item.id.

Please follow https://github.com/deployd/deployd/issues/292